RELEASES.md

```diff
@@ -0,0 +1,78 @@
+# Releases Guide
+
+Here, we will briefly understand the release plan and versioning steps. 
+This will cover the release process, major and minor releases, backporting the PR and manual release steps.
+
+If there is something that you require or this document leaves out, please
+reach out by [filing an issue](https://github.com/coredns/coredns/issues).
+
+## Releases
+
+Releases will be created from the release branches. The release will follow the following convention as `<major>.<minor>.<patch>`. 
+All the major and minor releases will be made from the master. The tag will be of the format `v<major>.<minor>.<patch>`.
+
+After a minor release, a branch will be created with the format of `release-<major>.<minor>` from the minor tag. Once the branch is ready, all the patches will be released from that branch. 
+For example, once we release `v1.10.0`, a branch will be created as `release-1.10` with the following `v1.10` tag. Next, all future patches will be done against that branch. i.e. `v1.10.0, v1.10.1`.
+
+## Next Release
+
+The next release will be tracked in the milestone under the issue section, with Release tracking. 
+Also, it can be filtered with the label release. If your issue or PR is not present in a milestone, please create a PR to track the next release.
+
+## Coredns Support
+
+The support will be identified with several states:
+
+- __*Active*__: The release is a stable branch that is currently supported and accepting patches.
+- __*Maintenance*__: The release that is currently supported and accepting patches.
+- __*End of Life*__: The release branch is no longer supported, and no new patches will be accepted.
+
+Currently, we are planning to support at least two latest major and minor releases **(N, N-1)**. The latest releases can be any combination of major or minor releases. Examples: a major release followed by two or more minor releases (e.g. 1.9, 1.10). 
+It is unknown when the next release will be made available. So, until then, the current or latest release will be considered the Active release.
+
+For e.g.
+
+The active release is *v1.10 (N)* and *v1.9 (N-1)* is in maintenance. But when the next release rolls out i.e. *v1.11 (N+1)* then *N, N+1* will be the active release, and *N-1* will be in maintenance mode.
```

Comment:
The wording is confusing to me, but I think this attempts to explain a common sense aspect of release progression that doesn't warrant spelling out.

```suggestion
```